src/App.tsx

```diff
@@ -1,24 +1,150 @@
-/* eslint-disable max-len */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
-import { UserWarning } from './UserWarning';
+import React, {
+  useCallback,
+  useEffect,
+  useMemo,
+  useState,
+} from 'react';
+import { Todo } from './types/Todo';
+import { TodoStatus } from './types/TodoStatus';
+import { ErrorMessage } from './types/ErrorMessage';
+import { deleteTodos, getTodos } from './api/todos';
+import { Header } from './components/Header/Header';
+import { TodoList } from './components/TodoList';
+import { Filter } from './components/Filter';
+import { Error } from './components/Error';
+import { USER_ID } from './types/ConstantTypes';
 
-const USER_ID = 0;
+function filterTodos(todos: Todo[], selectedFilter: TodoStatus): Todo[] {
+  return todos.filter((todo) => {
+    switch (selectedFilter) {
+      case TodoStatus.All:
+        return true;
+
+      case TodoStatus.Active:
+        return !todo.completed;
+
+      case TodoStatus.Completed:
+        return todo.completed;
+
+      default:
+        return true;
+    }
+  });
+}
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
-    return <UserWarning />;
-  }
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [selectedFilter, setSelectedFilter] = useState(TodoStatus.All);
+  const [errorType, setErrorType] = useState(ErrorMessage.None);
+  const [isErrorShown, setIsErrorShown] = useState(false);
+  const [query, setQuery] = useState('');
+  const [isClearCompleted, setIsClearCompleted] = useState(false);
+
+  useEffect(() => {
+    getTodos(USER_ID)
+      .then((userTodos) => setTodos(userTodos))
+      .catch(() => {
+        setErrorType(ErrorMessage.Download);
+        setIsErrorShown(true);
+      });
+  }, []);
+
+  const counterActiveTodos = useMemo(
+    () => todos.reduce((num, todo) => {
+      return todo.completed ? num : num + 1;
+    }, 0),
+    [todos],
+  );
+
+  const counterCompletedTodos = todos.length - counterActiveTodos;
+
+  const filteredTodos = useMemo(
+    () => filterTodos(todos, selectedFilter),
+    [selectedFilter, todos],
+  );
+
+  const showError = useCallback((error: ErrorMessage) => {
+    setErrorType(error);
+    setIsErrorShown(true);
+  }, []);
+
+  const hideError = useCallback(() => {
+    setIsErrorShown(false);
+  }, []);
+
+  const AddTodo = useCallback(
+    (newTodo: Todo): void => setTodos((oldTodos) => [...oldTodos, newTodo]),
+    [],
+  );
+
+  const DeleteTodo = useCallback(
+    (todoId: number): void => (
+      setTodos((oldTodos) => oldTodos.filter(todo => todo.id !== todoId))
+    ),
+    [],
+  );
+
+  const onClearCompleted = useCallback(() => {
```

Comment:
You disable this button
![image](https://github.com/mate-academy/react_todo-app-add-and-delete/assets/91411692/2274b347-1a28-4362-8c9d-611ee25ec9df)

Reply:
this button will be implemented in the next task